api/PclusterApiHandler.py

```diff
@@ -545,16 +557,15 @@ def get_identity():
         return {"cognito:groups": ["user", "admin"], "username": "username", "attributes": {"email": "user@domain.com"}}
 
     access_token = request.cookies.get("accessToken")
+    id_token = request.cookies.get("idToken")
     if not access_token:
         return {"message": "No access token."}, 401
     try:
-        decoded = jwt_decode(access_token, USER_POOL_ID)
+        decoded = jwt_decode(access_token)
+        decoded_id = jwt_decode(id_token, audience=AUDIENCE, access_token=access_token)
         username = decoded.get("username")
         if username:
-            cognito = boto3.client("cognito-idp")
-            filter_ = f'username = "{username}"'
-            user = cognito.list_users(UserPoolId=USER_POOL_ID, Filter=filter_)["Users"][0]
-            decoded["attributes"] = {ua["Name"]: ua["Value"] for ua in user["Attributes"]}
+            decoded["attributes"] = {key: value for key, value in decoded_id.items()}
```

Comment:
Here we don't need the check for the username, since this is related to the previous implementation based strongly on Cognito.
We could just have what is now on line 568 withouth the retrieval of the username and without the `if username`.